timesketch/api/v1/resources/graph.py

```diff
@@ -34,19 +39,90 @@ class GraphListResource(resources.ResourceMixin, Resource):
     """Resource to get all graphs."""
 
     @login_required
-    def get(self):
+    def get(self, sketch_id):
         """Handles GET request to the resource.
 
         Returns:
             List of graphs in JSON (instance of flask.wrappers.Response)
         """
-        graphs = manager.GraphManager.get_graphs()
-        graph_names = {name: graph_class().DISPLAY_NAME for name, graph_class in
-                       graphs}
-        return jsonify(graph_names)
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, 'No sketch found with this ID.')
+
+        graphs = sketch.graphs
+        return self.to_json(graphs)
+
+    def post(self, sketch_id):
+        """Handles POST request to the resource."""
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, 'No sketch found with this ID.')
```

Comment:
OK, I would like to see a check here to see if you have write access to the sketch, since we are writing a graph to the database... please make sure you are not allowing everyone with read access to add this here

Reply:
Done